### ISSUE TITLE
RUCSS creates a console error when used with Revolution Slider

### ISSUE THREAD

**Issue by DahmaniAdame** [wp-media/wp-rocket#3962]
**Before submitting an issue please check that you’ve completed the following steps:**
- [x] Made sure you’re on the latest version
- [x] Used the search feature to ensure that the bug hasn’t been reported before

**Describe the bug**
RUCSS removes an inline CSS script:

`<style id='rs-plugin-settings-inline-css' type='text/css'> #rs-demo-id {} </style>`

That RevSlider targets using another inline JS:

```
<script>
var htmlDivCss = unescape("");
var htmlDiv = document.getElementById('rs-plugin-settings-inline-css');
if (htmlDiv) {
    htmlDiv.innerHTML = htmlDiv.innerHTML + htmlDivCss;
} else {
    var htmlDiv = document.createElement('div');
    htmlDiv.innerHTML = '<style>' + htmlDivCss + '</style>';
    document.getElementsByTagName('head')[0].appendChild(htmlDiv.childNodes[0]); 
}
</script>
```

That particular line doesn't seem to play well when the corresponding ID `rs-plugin-settings-inline-css` is not there.

`document.getElementsByTagName('head')[0].appendChild(htmlDiv.childNodes[0]);`

**To Reproduce**
1. Add a Revolution Slider on the page
2. Enable RUCSS
3. See the error on the browser's console

**Expected behavior**
No error should be on the console resulting from using RUCSS. 

**Screenshots**
![image](https://user-images.githubusercontent.com/1289174/119939096-3a069e80-bf85-11eb-8796-51faf3c62202.png)

**Additional context**
Preserving `<style id='rs-plugin-settings-inline-css' type='text/css'> #rs-demo-id {} </style>` fixes the issue.

Ticket - https://secure.helpscout.net/conversation/1520234638/265876/
Slack discussion - https://wp-media.slack.com/archives/C43T1AYMQ/p1622116284103300

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

**Comment by DahmaniAdame** [comment]
Ticket - https://secure.helpscout.net/conversation/1520234638/265876/ 
Slack discussion - https://wp-media.slack.com/archives/C43T1AYMQ/p1622116284103300

**Comment by piotrbak** [comment]
@DahmaniAdame Just an addition that was found by @Mai-Saad 

Not all sliders made by Rev Slider cause this faulty script to be added to the source. If it's not added, then there's no error in the JS console. However, preserving this style in the source seems to be the best solution, just to be 100% safe.

**Comment by alfonso100** [comment]
Another case: https://secure.helpscout.net/conversation/1529596348/268184/

**Comment by alfonso100** [comment]
Another case: https://secure.helpscout.net/conversation/1556155336/275128?folderId=2683093

**Comment by mdarif** [comment]
+1

What's the latest here?

**Comment by NataliaDrause** [comment]
Related: https://secure.helpscout.net/conversation/1643296446/297416

**Comment by NataliaDrause** [comment]
Related: https://secure.helpscout.net/conversation/1586750255/283985/

**Comment by vmanthos** [comment]
Related: https://secure.helpscout.net/conversation/1647940834/298496?folderId=2135277

**Comment by vmanthos** [comment]
Related: https://secure.helpscout.net/conversation/1654470903/300069?folderId=2135277

**Comment by vmanthos** [comment]
**Related:** https://secure.helpscout.net/conversation/1680524126/304740/

In this case, apart from the JavaScript errors, the following issue was taking place:
![](https://i.vgy.me/IXTFYR.jpg)

This is happening because the CSS that's responsible for how the arrows are displayed is stored in JavaScript:
```JavaScript
var htmlDivCss = unescape(".dotdigital.tparrows%20%7B%0A%20%20%20%20text-decoration%3A%20none%3B%0A%20%20%20%20display%3A%20block%3B%0A%20%20%20%20width%3A%2080px%3B%0A%20%20%20%20height%3A%2080px%3B%0A%20%20%20%20margin%3A%20-40px%200%200%3B%0A%20%20%20%20position%3A%20absolute%3B%0A%20%20%20%20border-radius%3A%200%2050%25%2050%25%200%3B%0A%20%20%20%20top%3A%2050%25%3B%0A%20%20%20%20left%3A%20-40px%20%21important%3B%0A%20%20%20%20opacity%3A%201%3B%0A%20%20%20%20cursor%3A%20pointer%3B%0A%20%20%20%20font-size%3A%200%3B%0A%20%20%20%20color%3A%20%23ffffff%3B%0A%20%20%20%20background-color%3A%20rgba%28255%2C%20255%2C%20255%2C%200.1%29%3B%0A%20%20%20%20text-align%3A%20center%3B%0A%20%20%20%20transform%3A%20none%20%21important%3B%0A%20%20%20%20transition%3A%20all%200.3s%20ease-in-out%200s%3B%0A%7D%0A%0A%40media%20%28max-width%3A%20768px%29%20%7B%0A%20%20.dotdigital.tparrows%20%7B%0A%20%20%20%20%20%20top%3A%2040%25%20%21important%3B%0A%20%20%7D%0A%7D%0A%0A.dotdigital.tparrows.tp-rightarrow%20%7B%0A%20%20%20%20left%3A%20auto%20%21important%3B%0A%20%20%20%20right%3A%20-40px%20%21important%3B%0A%20%20%20%20border-radius%3A%2050%25%200%200%2050%25%3B%0A%7D%0A.dotdigital.tparrows%3Abefore%20%7B%0A%20%20%20%20position%3A%20absolute%3B%0A%20%20%20%20top%3A%2020px%3B%0A%20%20%20%20left%3A%20auto%3B%0A%20%20%20%20right%3A%2014px%3B%0A%20%20%20%20transition%3A%20all%200.3s%20ease-in-out%200s%3B%0A%7D%0A.dotdigital.tparrows.tp-rightarrow%3Abefore%20%7B%0A%20%20%20%20right%3A%20auto%3B%0A%20%20%20%20left%3A%2014px%3B%0A%7D%0A%0A.dotdigital.tparrows%3Ahover%20%7B%0A%20%20%20%20background-color%3A%20%23ff497c%3B%0A%20%20%20%20left%3A%200%20%21important%3B%0A%7D%0A%0A.dotdigital.tparrows.tp-rightarrow%3Ahover%20%7B%0A%20%20%20%20left%3A%20auto%20%21important%3B%0A%20%20%20%20right%3A%200%20%21important%3B%0A%7D%0A%0A.dotdigital.tparrows%3Ahover%3Abefore%20%7B%0A%20%20%20%20left%3A%20auto%3B%0A%20%20%20%20right%3A%2040px%3B%0A%7D%0A%0A.dotdigital.tparrows.tp-rightarrow%3Ahover%3Abefore%20%7B%0A%20%20%20%20left%3A%2040px%3B%0A%20%20%20%20right%3A%20auto%3B%0A%7D%0A");
```
This CSS should be injected to `<style id='rs-plugin-settings-inline-css' type='text/css'> #rs-demo-id {} </style>` that @DahmaniAdame mentioned in the original post.

A temporary workaround is to use the following:

```PHP
add_action( 'wp_head', function() {
	
	if( ! function_exists('get_rocket_option') || 0 == get_rocket_option( 'remove_unused_css' ) ) return;
 
	$css = '<style id="rs-plugin-settings-inline-css" type="text/css">#rs-demo-id{}.dotdigital.tparrows{text-decoration: none; display: block; width: 80px; height: 80px; margin: -40px 0 0; position: absolute; border-radius: 0 50% 50% 0; top: 50%; left: -40px !important; opacity: 1; cursor: pointer; font-size: 0; color: #ffffff; background-color: rgba(255, 255, 255, 0.1); text-align: center; transform: none !important; transition: all 0.3s ease-in-out 0s;}@media (max-width: 768px){.dotdigital.tparrows{top: 40% !important;}}.dotdigital.tparrows.tp-rightarrow{left: auto !important; right: -40px !important; border-radius: 50% 0 0 50%;}.dotdigital.tparrows:before{position: absolute; top: 20px; left: auto; right: 14px; transition: all 0.3s ease-in-out 0s;}.dotdigital.tparrows.tp-rightarrow:before{right: auto; left: 14px;}.dotdigital.tparrows:hover{background-color: #ff497c; left: 0 !important;}.dotdigital.tparrows.tp-rightarrow:hover{left: auto !important; right: 0 !important;}.dotdigital.tparrows:hover:before{left: auto; right: 40px;}.dotdigital.tparrows.tp-rightarrow:hover:before{left: 40px; right: auto;}</style>';

	if( function_exists( 'is_front_page' ) && is_front_page() ) echo $css;
});
```

This inserts the previously mentioned CSS in the page, and that allows the used CSS generator to process it and add-it in the used CSS.

**Comment by piotrbak** [comment]
Fixed with 3.11:
https://github.com/wp-media/wp-rocket/blob/63f649200fa3b5b81ec83812d3448eeeea28e464/inc/Engine/Optimization/RUCSS/Controller/UsedCSS.php#L67
